### PR TITLE
add additional typecheck to Discovery/renderAggregation

### PR DESCRIPTION
**Pull request by craigrbarnes**
Jira Ticket: [PXP-11229](https://ctds-planx.atlassian.net/browse/PXP-11229)

### Bug Fixes
Adds an additional check to DiscoverySummary `renderAggregation` to only sum strings or number values.  If the value is something other than a string or number, an array, for example, the sum value explodes.

### Deployment changes

[PXP-11229]: https://ctds-planx.atlassian.net/browse/PXP-11229?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/Discovery/DiscoverySummary.tsx [hunk 1]
```diff
@@ -21,7 +21,16 @@ const renderAggregation = (aggregation: AggregationConfig, studies: any[] | null
   switch (type) {
   case 'sum': {
     // parse any string representation of an integer
-    fields = fields.map((item) => (typeof item === 'string' ? parseInt(item, 10) || 0 : item));
+    // if it's not a number, return 0 so as not to break the sum
```
Comment (jarvisraymond-uchicago) [inline src/Discovery/DiscoverySummary.tsx:24]: Nitpick: instead of using a comment here, the code could be wrapped in a function with an explanatory name. Reduces likelihood that code changes and the comment does not.  

Examples: 
```
const checkForNonNumericInput = (input) => {
  ....
}
```
or
```
const handleNonNumericInput = () => {
   ...
}
```

File: src/Discovery/DiscoverySummary.tsx [hunk 2]
```diff
@@ -21,7 +21,16 @@ const renderAggregation = (aggregation: AggregationConfig, studies: any[] | null
...
+    fields = fields.map((item) => {
+      if (typeof item === 'number') {
+        return item;
+      }
+      if (typeof item === 'string') {
```
Comment (jarvisraymond-uchicago) [inline src/Discovery/DiscoverySummary.tsx:29]: Nitpick: changing this to an ```else if``` is more efficient and reduces number of comparisons executed (but practically this doesn't matter).

File: src/Discovery/DiscoverySummary.tsx [hunk 3]
```diff
@@ -21,7 +21,16 @@ const renderAggregation = (aggregation: AggregationConfig, studies: any[] | null
...
+        return parseInt(item, 10) || 0;
```
Comment (jarvisraymond-uchicago) [inline src/Discovery/DiscoverySummary.tsx:30]: Nitpick: I believe the second parameter for parseInt sets the base, however the default for parseInt is base 10 so the second parameter isn't needed.

**Review by jarvisraymond-uchicago — APPROVED** [review]
Great catch! This will hopefully reduce runtime errors! 

* Left some nitpick comments, update if deemed necessary. 
* Please add JIRA ticket to PR description using format: 
```Jira Ticket: [VADC-764](https://ctds-planx.atlassian.net/browse/VADC-764)```

**Comment by mfshao** [comment]
what is the relationship between this PR and VADC-764?

**Comment by jarvisraymond-uchicago** [comment]
> what is the relationship between this PR and VADC-764?

None, however this is the format needed for PR's
* Please add JIRA ticket to PR description using format:
Jira Ticket: [VADC-764](https://ctds-planx.atlassian.net/browse/VADC-764)

**Review by mfshao — COMMENTED** [review]
trying to clarify the goal here: if the target field are arrays of strings or numbers, shouldn't the expected behavior of this `sum` function is to sum the value of all numeric elements in those arrays?

**Comment by craigrbarnes** [comment]
> trying to clarify the goal here: if the target field are arrays of strings or numbers, shouldn't the expected behavior of this `sum` function is to sum the value of all numeric elements in those arrays?

True, I can add support for arrays as well.

File: src/Discovery/DiscoverySummary.tsx (hunk 1 above)
Comment (craigrbarnes) [reply src/Discovery/DiscoverySummary.tsx]: used your first suggestion

File: src/Discovery/DiscoverySummary.tsx (hunk 2 above)
Comment (craigrbarnes) [reply src/Discovery/DiscoverySummary.tsx]: Since I added handling of an array and it's a return. I left it as 3 if..then blocks

File: src/Discovery/DiscoverySummary.tsx (hunk 3 above)
Comment (craigrbarnes) [reply src/Discovery/DiscoverySummary.tsx]: It is but ESlint wants to make it required

File: package-lock.json [hunk 4]
```diff
@@ -61625,7 +61625,7 @@
         "fb-watchman": "^2.0.0",
         "fbjs": "^3.0.0",
         "glob": "^7.1.1",
-        "immutable": "*",
+        "immutable": "~3.7.6",
```
Comment (jarvisraymond-uchicago) [inline package-lock.json:61628]: Is this change intentional? If not this file should be reverted. 

If you want to revert the file to its state in master in git:

```git checkout origin/master [filename]```

Comment (craigrbarnes) [reply package-lock.json]: It was not, I will do that now.

**Review by jarvisraymond-uchicago — APPROVED** [review]

**Review by mfshao — APPROVED** [review]

**Comment by mfshao** [comment]
those failed CI tests doesn't look like related to this PR
